node/impl/full/chain.go

```diff
@@ -654,6 +656,33 @@ func (a *ChainAPI) ChainBlockstoreInfo(ctx context.Context) (map[string]interfac
 	return info.Info(), nil
 }
 
+// ChainGetEvents returns the events under an event AMT root CID.
+//
+// TODO (raulk) make copies of this logic elsewhere use this (e.g. itests, CLI, events filter).
```

Comment:
Yes please.

Reply:
All of the events logic is about to change with https://github.com/filecoin-project/ref-fvm/issues/1376, so this will have to get refactored. For context, ref-fvm no longer writes events to the store and instead just returns them in ApplyRet.